src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -496,103 +467,102 @@ namespace SurfaceGeometry {
 
         for (int ZoneNum = 1; ZoneNum <= state.dataGlobal->NumOfZones; ++ZoneNum) {
             auto &thisZone = state.dataHeatBal->Zone(ZoneNum);
+            int CeilCount = 0.0;
+            int FloorCount = 0.0;
+            int WallCount = 0;
+            Real64 AverageHeight = 0.0; // Used to keep track of average height of a surface/zone
+            Real64 ZMax = -99999.0;     // Maximum Z of a surface (detailed outside coefficient calculation)
+            Real64 ZMin = 99999.0;      // Minimum Z of a surface (detailed outside coefficient calculation)
+            Real64 ZCeilAvg = 0.0;
+            Real64 ZFlrAvg = 0.0;
+            if (DetailedWWR) {
+                print(state.files.debug, "{},{:.2R},{:.2R}\n", thisZone.Name, thisZone.ExtGrossWallArea, thisZone.ExtWindowArea);
+            }
             for (int spaceNum : thisZone.spaceIndexes) {
                 auto &thisSpace = state.dataHeatBal->space(spaceNum);
-                CeilCount = 0.0;
-                FloorCount = 0.0;
-                Count = 0;
-                AverageHeight = 0.0;
-                ZCeilAvg = 0.0;
-                ZFlrAvg = 0.0;
-                ZMax = -99999.0;
-                ZMin = 99999.0;
-                if (DetailedWWR) {
-                    print(state.files.debug, "{},{:.2R},{:.2R}\n", thisZone.Name, thisZone.ExtGrossWallArea, thisZone.ExtWindowArea);
-                }
                 // Use AllSurfaceFirst which includes air boundaries
                 for (int SurfNum = thisSpace.AllSurfaceFirst; SurfNum <= thisSpace.AllSurfaceLast; ++SurfNum) {
                     auto &thisSurface = state.dataSurface->Surface(SurfNum);
 
                     if (thisSurface.Class == SurfaceClass::Roof) {
                         // Use Average Z for surface, more important for roofs than floors...
                         ++CeilCount;
-                        Z1 = minval(thisSurface.Vertex({1, thisSurface.Sides}), &Vector::z);
-                        Z2 = maxval(thisSurface.Vertex({1, thisSurface.Sides}), &Vector::z);
+                        Real64 Z1 = minval(thisSurface.Vertex({1, thisSurface.Sides}), &Vector::z);
```

Comment:
Why slice here? Isn't this the entire vertex array?

Reply:
> Why slice here? Isn't this the entire vertex array?

Looks like this was done to support a certain compiler back in the day and just carried forward through all the conversions. https://github.com/NREL/EnergyPlusArchive/commit/dc94f2092324ca6d039e4699305a9c4200348664

I'll try changing all of these wherever they may show up.

Reply:
Yes please. Slices create copies and we should avoid them to the extent possible. They're a convenient shorthand perhaps but quite inefficient.